lint/src/main/java/org/datacommons/lint/Lint.java

```diff
@@ -0,0 +1,162 @@
+package org.datacommons.lint;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.Callable;
+import org.datacommons.proto.Debug;
+import org.datacommons.proto.Mcf;
+import org.datacommons.util.McfParser;
+import org.datacommons.util.McfUtil;
+import org.datacommons.util.TmcfCsvParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+    name = "DCLint",
+    mixinStandardHelpOptions = true,
+    version = "DCLint 0.1",
+    description = "Tool for use in developing datasets for Data Commons.")
+class Lint implements Callable<Integer> {
+  private static final Logger logger = LoggerFactory.getLogger(Lint.class);
+
+  @CommandLine.Parameters(
+      arity = "1..*",
+      description =
+          ("List of input files. This depends on the --format option. "
+              + "For 'tmcf', the files are expected to be "
+              + "Template MCF files. For 'mcf', the files are expected to be Instance MCF "
+              + "files. For tmcfCsv, the first file should be a Template MCF file, followed "
+              + "by one or more CSVs compatible with the provided TMCF file."))
+  private File[] files;
+
+  enum FormatType {
+    mcf,
+    tmcf,
+    tmcfCsv
+  };
+
+  @CommandLine.Option(
+      names = {"-f", "--format"},
+      description = "Format of input files: ${COMPLETION-CANDIDATES}",
+      required = true)
+  private FormatType formatType;
+
+  enum CommandType {
+    chk,
+    genMcf
+  };
+
+  @CommandLine.Option(
+      names = {"-c", "--cmd"},
+      description =
+          "The command to run: ${COMPLETION-CANDIDATES}.  'chk' runs "
+              + "syntax checks on the input files.  'genMcf' produces instance MCF output, "
+              + "and only makes sense when --format is 'tmcfCsv'. Default: 'chk'",
+      defaultValue = "chk",
+      required = true)
+  private CommandType cmdType;
+
+  @CommandLine.Option(
+      names = {"-d", "--delimiter"},
+      defaultValue = ",",
+      description =
+          "Delimiter of the input CSV file. Relevant only when --format is 'tmcfCsv'. "
+              + "Default: ',' (comma)")
+  private char delimiter;
+
+  @CommandLine.Option(
+      names = {"-o", "--outputDir"},
+      description = "Directory to write output files. Default: /tmp",
+      defaultValue = "/tmp",
+      scope = CommandLine.ScopeType.INHERIT)
+  private File outputDir;
+
+  @CommandLine.Option(
+      names = {"-r", "--isResolved"},
```

Comment:
I've simplified it for now to be unresolved.

But it is useful to know what the user intended to do, provide resolved MCF or unresolved MCF, so that we can check accordingly. For instance, if they intended to provide resolved MCF and we find a local reference that's an error. But if we find the same with unresolved MCF, that's fine.

Reply:
I see that makes sense. "resolved" might be a concept that user gets confused (and they may give wrong option), hopefully we will have a way to hide this in the future..